vyper/semantics/types/function.py

```diff
@@ -700,7 +726,7 @@ def _parse_return_type(funcdef: vy_ast.FunctionDef) -> Optional[VyperType]:
 
 def _parse_decorators(
     funcdef: vy_ast.FunctionDef,
-) -> tuple[FunctionVisibility, StateMutability, bool]:
+) -> tuple[FunctionVisibility | None, StateMutability, bool]:
```

Comment:
nit: prefer `Optional[FunctionVisibility]`

Reply:
addressed in: https://github.com/vyperlang/vyper/pull/4178/commits/e50e44e3d39e9e1d3d525a4922f56589f9e6bfbc